inc/Engine/Optimization/DelayJS/HTML.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace WP_Rocket\Engine\Optimization\DelayJS;
+
+use WP_Rocket\Admin\Options_Data;
+
+class HTML {
+
+	/**
+	 * Plugin options instance.
+	 *
+	 * @since  3.7
+	 *
+	 * @var Options_Data
+	 */
+	protected $options;
+
+	/**
+	 * Allowed scripts regex.
+	 *
+	 * @since 3.7
+	 *
+	 * @var string
+	 */
+	private $allowed_scripts = '';
+
+	/**
+	 * Creates an instance of HTML.
+	 *
+	 * @since  3.7
+	 *
+	 * @param Options_Data $options Plugin options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Adjust HTML to have delay js structure.
+	 *
+	 * @param string $html Buffer html for the page.
+	 *
+	 * @return string
+	 */
+	public function delay_js( $html ) {
+
+		if ( ! $this->is_allowed() ) {
+			return $html;
+		}
+
+		$this->allowed_scripts = $this->prepare_allowed_scripts_regex();
+
+		return $this->parse( $html );
+	}
+
+	/**
+	 * Checks if is allowed to Delay JS.
+	 *
+	 * @since 3.7
+	 *
+	 * @return bool
+	 */
+	public function is_allowed() {
+		if ( rocket_bypass() ) {
+			return false;
+		}
+
+		if ( rocket_get_constant( 'DONOTROCKETOPTIMIZE' ) ) {
+			return false;
+		}
+
+		if ( rocket_get_constant( 'DONOTDELAYJS' ) ) {
+			return false;
+		}
+
+		if ( ! $this->options->get( 'delay_js' ) ) {
+			return false;
+		}
+
+		return true;
```

Comment:
This can be simplified to
```suggestion
		return (bool) $this->options->get( 'delay_js', 0 ) );
```